service/collector/elb.go

```diff
@@ -0,0 +1,136 @@
+package collector
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	AccountLabel = "account"
+)
+
+var (
+	elbsDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "elb_instance_out_of_service_count"),
+		"Gauge about ELB instances being out of service.",
+		[]string{
+			AccountLabel,
+			ClusterLabel,
+			InstallationLabel,
+			OrganizationLabel,
+		},
+		nil,
+	)
+)
+
+func (c *Collector) collectAccountsELBs(ch chan<- prometheus.Metric, clients []aws.Clients) {
+	var wg sync.WaitGroup
+
+	for _, client := range clients {
+		wg.Add(1)
+		go func(awsClients aws.Clients) {
+			defer wg.Done()
+			err := c.collectELBs(ch, awsClients)
+			if err != nil {
+				c.logger.Log("level", "error", "message", "failed collecting ELB metrics", "stack", fmt.Sprintf("%#v", err))
+			}
+		}(client)
+	}
+
+	wg.Wait()
+}
+
+func (c *Collector) collectELBs(ch chan<- prometheus.Metric, awsClients aws.Clients) error {
+	c.logger.Log("level", "debug", "message", "collecting metrics for elbs")
+
+	account, err := c.awsAccountID(awsClients)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var loadbalancers []*elb.LoadBalancerDescription
+	{
+		i := &elb.DescribeLoadBalancersInput{}
+		o, err := awsClients.ELB.DescribeLoadBalancers(i)
```

Comment:
I was trying to use the paging API but this was a bit unfortunate in the code. I think it is good enough because we have less than 1000 ELBs in an account. The number should rather be 50 at max or something.

Reply:
Each page is 1000 ELBs?

Reply:
The default is 400. You can page with 2, but I do not see the point right now.